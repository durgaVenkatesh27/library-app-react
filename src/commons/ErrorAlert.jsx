import React from 'react';
import PropTypes from 'prop-types';
import Alert from '@material-ui/lab/Alert';
import '../index.css';

const ErrorAlert = ({ errorMessage, errorSeverity }) => (
  <Alert severity={errorSeverity} className="custom-alert">{errorMessage}</Alert>
);

ErrorAlert.propTypes = {
  errorMessage: PropTypes.string.isRequired
};

ErrorAlert.defaultProps = {
  errorMessage: '',
  errorSeverity: 'error'
};
export default ErrorAlert;