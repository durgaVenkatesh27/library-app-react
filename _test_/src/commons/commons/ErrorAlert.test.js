import React from 'react';
import { shallow } from 'enzyme';
import ErrorAlert from '../../../../src/commons/ErrorAlert';


describe('Error Alert Tests', () => {
    let props;
    beforeEach(() => {
        props = {
            errorMessage: "Some error!!!",
            errorSeverity: 'error'
        }
    });

    it('should render Library app error Component', () => {
        const renderedModule = shallow(< ErrorAlert {...props}
        />);
        expect(renderedModule).toMatchSnapshot(); expect(renderedModule.find('.custom-alert').length).toBe(1); expect(renderedModule.find('.custom-alert').text()).toEqual('Some error!!!');
    });
});