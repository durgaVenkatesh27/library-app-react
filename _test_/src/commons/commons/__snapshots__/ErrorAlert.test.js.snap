// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Alert Tests should render Library app error Component 1`] = `ShallowWrapper {}`;
